clang/lib/CIR/CodeGen/CIRGenRecordLayoutBuilder.cpp

```diff
@@ -126,6 +130,12 @@ void CIRRecordLowering::lower() {
     return;
   }
 
+  if (isa<CXXRecordDecl>(recordDecl)) {
+    cirGenTypes.getCGModule().errorNYI(recordDecl->getSourceRange(),
```

Comment:
Oh my!  This ends up being harsher than I expected!  This means no record-types in C++ at all (even ones that don't do anything).  But if that is the level of support we have, I'm ok with it.

I presume a soon-to-be used patch will reduce this to '!virtual && !getNumBases` etc.

Reply:
Eventually, yes. We aren't handling CXXRecordDecl in the code that gets here yet. I don't expect there to be much more than adding a switch statement for that, but everything in C++ ends up being more involved than I realized, so who knows.